### PR TITLE
Fix usage of complex types

### PR DESCRIPTION
**Pull request by castigli**
so far we have treated complex and double pointers the same way, this PR introduces explicit usage of `complex<>`

### CODE REVIEW

File: include/Quiccir/Transforms/Utils.h [hunk 1]
```diff
@@ -49,24 +49,63 @@ getLibraryCallSymbolRef(Operation *op, PatternRewriter &rewriter, ArrayRef<Type>
   // Return types
   for (Type ret : op->getResultTypes()) {
     if (auto tensor = ret.dyn_cast<RankedTensorType>()) {
+      auto eleTy = tensor.getElementType();
```
Comment (phmarti) [inline include/Quiccir/Transforms/Utils.h]: Can't we replace those four blocks with a lambda or something?

File: lib/Quiccir/IR/QuiccirDialect.cpp [hunk 2]
```diff
@@ -289,6 +302,18 @@ mlir::LogicalResult FrIOp::verify() {
   auto physType = phys.getType().dyn_cast<RankedTensorType>();
   auto resultType = getType().dyn_cast<RankedTensorType>();
 
+  // Check element Types
```
Comment (phmarti) [inline lib/Quiccir/IR/QuiccirDialect.cpp:305]: Same here, replace by function call?

**Review by phmarti — APPROVED** [review]
Just a couple of comments.

File: include/Quiccir/Transforms/Utils.h (hunk 1 above)
Comment (castigli) [reply include/Quiccir/Transforms/Utils.h]: done

File: lib/Quiccir/IR/QuiccirDialect.cpp (hunk 2 above)
Comment (castigli) [reply lib/Quiccir/IR/QuiccirDialect.cpp:305]: this is a bit trickier to do cleanly, the expected types depend on the op and the phys/mods name depends on the direction

**Pull request merged**